cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -204,15 +203,21 @@ def upload_rna_report_to_dna_case_in_scout(
         report_file: File,
         rna_case_id: str,
     ) -> None:
-        """Upload report file to DNA cases related to a RNA case in scout."""
+        """Upload report file to DNA cases related to an RNA case in scout."""
         LOG.info(f"Finding DNA cases related to RNA case {rna_case_id}")
-        for dna_case_id in self.get_unique_dna_cases_related_to_rna_case(rna_case_id):
-            self.upload_report_to_scout(
-                dry_run=dry_run,
-                report_type=report_type,
-                report_file=report_file,
-                case_id=dna_case_id,
-            )
+        uploaded: bool = False
+        for dna_case_id in self.get_unique_dna_cases_related_to_rna_case(case_id=rna_case_id):
+            dna_case: Family = self.status_db.get_case_by_internal_id(internal_id=dna_case_id)
+            if dna_case.analyses and dna_case.analyses[0].uploaded_at:
+                self.upload_report_to_scout(
+                    dry_run=dry_run,
+                    report_type=report_type,
+                    report_file=report_file,
+                    case_id=dna_case_id,
+                )
+                uploaded = True
+        if not uploaded:
+            raise CgDataError("No connected DNA case has been uploaded.")
```

Comment:
Should we try-except this error in the caller and proceed or raise and exit?

Reply:
Checked with @karlnyr and if no DNA case has been uploaded, we quit, but if some were successful, we upload to them. Ran another test for this scenario.